crates/storage/src/mockdb.rs

```diff
@@ -641,9 +642,46 @@ impl DB for MockDB {
         _cf: &DbColFam,
         _key: &Key,
         _new_value: impl AsRef<[u8]>,
+        _persist_diffs: bool,
     ) -> Result<()> {
         unimplemented!()
     }
+
+    fn migrator() -> Self::Migrator {
+        unimplemented!("Migration isn't implemented in MockDB")
+    }
+
+    fn update_last_block_merkle_tree(
+        &self,
+        merkle_tree_stores: MerkleTreeStoresWrite<'_>,
+    ) -> Result<()> {
+        // Read the last block's height
+        let height: BlockHeight = self.read_value(BLOCK_HEIGHT_KEY)?.unwrap();
+
+        // Read the last block's epoch
+        let prefix = height.raw();
+        let epoch_key = format!("{prefix}/{EPOCH_KEY_SEGMENT}");
+        let epoch: Epoch = self.read_value(epoch_key)?.unwrap();
+
+        for st in StoreType::iter() {
+            if st.is_stored_every_block() {
+                let key_prefix = if st.is_stored_every_block() {
+                    tree_key_prefix_with_height(st, height)
+                } else {
+                    tree_key_prefix_with_epoch(st, epoch)
+                };
```

Comment:
In this case, we can use always `tree_key_prefix_with_height`

Reply:
What about the case when the migration happens to run on a block during epoch change? Do we maybe also need to use `is_full_commit` for `update_last_block_merkle_tree` to correctly update the stores with `tree_key_prefix_with_epoch`?

Reply:
ah, yes, we need to update the other trees when the epoch change.

Reply:
added in [ffefc11](https://github.com/anoma/namada/pull/4135/commits/ffefc11da05b9977856e1c5cd7f8e8194904cf7c)